44.md

```diff
@@ -44,52 +44,16 @@ This encryption scheme replaces the one described in NIP-04, which is not secure
 ## Javascript
 
 ```javascript
-import {xchacha20} from "@noble/ciphers/chacha"
-import {secp256k1} from "@noble/curves/secp256k1"
-import {sha256} from "@noble/hashes/sha256"
-import {randomBytes, concatBytes} from "@noble/hashes/utils"
-import {base64} from "@scure/base"
-
-export const utf8Decoder = new TextDecoder()
-
-export const utf8Encoder = new TextEncoder()
-
-export type TLV = {[t: number]: Uint8Array[]}
-
-export function parseTLV(data: Uint8Array): TLV {
-  let result: TLV = {}
-  let rest = data
-  while (rest.length > 0) {
-    let t = rest[0]
-    let l = rest[1]
-    if (!l) throw new Error(`malformed TLV ${t}`)
-    let v = rest.slice(2, 2 + l)
-    rest = rest.slice(2 + l)
-    if (v.length < l) throw new Error(`not enough data to read on TLV ${t}`)
-    result[t] = result[t] || []
-    result[t].push(v)
-  }
-  return result
-}
-
-export function encodeTLV(tlv: TLV): Uint8Array {
-  let entries: Uint8Array[] = []
-
-  Object.entries(tlv).forEach(([t, vs]) => {
-    vs.forEach(v => {
-      let entry = new Uint8Array(v.length + 2)
-      entry.set([parseInt(t)], 0)
-      entry.set([v.length], 1)
-      entry.set(v, 2)
-      entries.push(entry)
-    })
-  })
+import {base64} from '@scure/base'
+import {randomBytes} from '@noble/hashes/utils'
+import {secp256k1} from '@noble/curves/secp256k1'
+import {sha256} from '@noble/hashes/sha256'
+import {xchacha20} from '@noble/ciphers/chacha'
 
-  return concatBytes(...entries)
-}
+import {utf8Decoder, utf8Encoder} from './utils.ts'
 
 export const getSharedSecret = (privkey: string, pubkey: string): Uint8Array =>
-  sha256(secp256k1.getSharedSecret(privkey, "02" + pubkey).subarray(1, 33))
+  sha256(secp256k1.getSharedSecret(privkey, '02' + pubkey).subarray(1, 33))
```

Comment:
My only concern with regards to the algorithms in the NIP is the fact we're always prepending `02` parity byte. Not sure about potential consequences of that. E.g. bias towards one, or another kind of shared keys.